tests/git.zunit

```diff
@@ -4299,6 +4299,213 @@
+        assert ${actual2[1]} same_as 'containing'
+
+        actual3=(${(0)lines[3]})
+        assert ${#actual3} equals 1
+        assert ${actual3[1]} same_as 'newlines'
+    }
+
+    prefix=
+    _fzf_complete_git 'git show another-branch -- '
+}
+
+@test 'Testing completion: git show another-branch:**' {
+    _fzf_complete() {
+        assert $# equals 5
+        assert $1 same_as '--ansi'
+        assert $2 same_as '--read0'
+        assert $3 same_as '--print0'
+        assert $4 same_as '--'
+        assert $5 same_as 'git show another-branch:'
+
+        run cat
+        assert ${#lines} equals 3
+
+        actual1=(${(0)lines[1]})
+        assert ${#actual1} equals 3
+        assert ${actual1[1]} same_as ' file3 containing space '
+        assert ${actual1[2]} same_as 'directory1/file2'
+        assert ${actual1[3]} same_as 'directory2/file4'
+
+        actual2=(${(0)lines[2]})
+        assert ${#actual2} equals 1
+        assert ${actual2[1]} same_as 'containing'
+
+        actual3=(${(0)lines[3]})
+        assert ${#actual3} equals 2
+        assert ${actual3[1]} same_as 'newlines'
+        assert ${actual3[2]} same_as 'file1'
+    }
+
+    prefix=another-branch:
+    _fzf_complete_git 'git show '
+}
+
+@test 'Testing completion: git show another-branch:file1 master another-branch v1:file1 -- **' {
+    _fzf_complete() {
+        assert $# equals 6
+        assert $1 same_as '--ansi'
+        assert $2 same_as '--read0'
+        assert $3 same_as '--print0'
+        assert $4 same_as '--multi'
+        assert $5 same_as '--'
+        assert $6 same_as 'git show another-branch:file1 master another-branch v1:file1 -- '
+
+        run cat
+        assert ${#lines} equals 3
+
+        actual1=(${(0)lines[1]})
+        assert ${#actual1} equals 3
+        assert ${actual1[1]} same_as ' file3 containing space '
+        assert ${actual1[2]} same_as 'directory1/file2'
+        assert ${actual1[3]} same_as 'directory2/file4'
+
+        actual2=(${(0)lines[2]})
+        assert ${#actual2} equals 1
+        assert ${actual2[1]} same_as 'containing'
+
+        actual3=(${(0)lines[3]})
+        assert ${#actual3} equals 2
+        assert ${actual3[1]} same_as 'newlines'
+        assert ${actual3[2]} same_as 'file1'
+    }
+
+    prefix=
+    _fzf_complete_git 'git show another-branch:file1 master another-branch v1:file1 -- '
+}
+
+@test 'Testing completion: git show --show-notes=**' {
+    run git notes add -m 'Note1'
+    run git notes --ref=note add -m 'Note2'
+
+    _fzf_complete() {
+        assert $# equals 4
+        assert $1 same_as '--ansi'
+        assert $2 same_as '--tiebreak=index'
+        assert $3 same_as '--'
+        assert $4 same_as 'git show --show-notes='
+
+        run cat
+        assert ${#lines} equals 2
+
+        assert ${lines[1]} same_as "${fg[yellow]}notes/commits$reset_color  Notes added by 'git notes add'"
+        assert ${lines[2]} same_as "${fg[yellow]}notes/note   $reset_color  Notes added by 'git notes add'"
```

Comment:
From out of curiousity, is it technically possible to show the content of notes in completion such as `Note1`?

Reply:
One notes ref is related with some commits. It is maybe difficult to display multiple notes message such as `Note1`.

Reply:
OK.